aiida_common_workflows/workflows/relax/siesta/generator.py

```diff
@@ -96,6 +97,13 @@ def get_builder(
             raise ValueError('Wrong relaxation type: no relax_type with name {} implemented'.format(relaxation_type))
         if 'relaxation' not in calc_engines:
             raise ValueError('The `calc_engines` dictionaly must contain "relaxation" as outermost key')
+        if previous_workchain:
+            try:
+                prev_wc_class = previous_workchain.process_class
+                if not prev_wc_class == self.process_class:
+                    raise ValueError('The "previous_workchain" must be a node of SiestaRelaxWorkChain')
+            except AttributeError:
+                raise ValueError('The "previous_workchain" must be a node of SiestaRelaxWorkChain')
```

Comment:
Should we maybe generalize this and put it in the abstract class? Won't all input generators not compare to `self.process_class`? If so, we might just implement this once and have all subclasses just call  `super`.

Reply:
Yes, now that you point it out, I see that is general enough. However we have to force each implementation to put the `super` call in their `get_builder`, which was not the case yet. I don't know...is it worth it?

Reply:
I think so, because otherwise they will more or less have to copy paste the 7 lines you added. Adding one line instead of seven is a win in my book.

Reply:
Ok, I will do it!